src/file-input/internal.tsx

```diff
@@ -47,7 +59,10 @@ const InternalFileInput = React.forwardRef(
 
     const [isFocused, setIsFocused] = useState(false);
     const onUploadButtonClick = () => uploadInputRef.current?.click();
-    const onUploadInputFocus = () => setIsFocused(true);
+    const onUploadInputFocus = () => {
+      setIsFocused(true);
+      containerRef.current?.scrollIntoView();
```

Comment:
nit: we can probably check if the scrollIntoView() function is defined so that we don't need to mock it in the unit tests